### PR TITLE
Break out core get FSM logic

### PR DESCRIPTION
**Pull request by jonmeredith**
This breaks out the core logic of the get FSM so it can be reused for modelling and other places.

It also fixes an edge case with deletion - now deletion only takes place if all vnodes respond with the same object and there are no read repairs.  Otherwise it is possible for vnodes to be an ancestor of the deleted object without the tombstone in the metadata.  If the delete is not processed on the ancestor vnodes, but succeeds on the tombstone vnodes the object would be resurrected (could happen on an out-of-disk-space bitcask node).

### CODE REVIEW

File: src/riak_kv_get_fsm.erl [hunk 1]
```diff
@@ -43,24 +43,15 @@
```
Comment (argv0) [inline src/riak_kv_get_fsm.erl:51]: ^^^ need to update the options() spec to include details?

**Comment by jonmeredith** [comment]
Just to be thorough :)

OK, passed 10000000 tests
true

**Comment by argv0** [comment]
experience quickcheck.

**Comment by argv0** [comment]
+1 to merge

**Comment by jonmeredith** [comment]
Also received +1 by email from Justin Sheehy.

**Pull request merged**